frontend/js/src/user-feed/UserFeed.tsx

```diff
@@ -680,7 +843,16 @@ export default function UserFeedPage() {
               >
                 <ul>
                   {events?.map((event) => {
-                    const { created, event_type, user_name } = event;
+                    const { created, event_type, user_name, metadata } = event;
+                    let subEventElement;
+                    if(event_type === EventType.THANKS && !event.hidden){
+                      const {original_event_id, original_event_type} = metadata as ThanksMetadata;
+                      const subEvent = events?.find(
+                        (evt) => evt.id === original_event_id && evt.event_type === original_event_type
+                      );
+                      subEventElement = renderSubEvent(subEvent);
+                    }
+
                     return (
                       <li
                         className="timeline-event"
```

Comment:
I can't add a comment below, but there is an attribute missing in the icon rendering:
Please add the `fixedWidth` attribute to both FontAwesomeIcons in the `event-description` div.
This will improve the horizontal positioning of the icons (all of them, not just the thank events):

![image](https://github.com/user-attachments/assets/76130dbd-176e-4ef7-9f5c-011db0249f30)

![image](https://github.com/user-attachments/assets/d8139cc9-a5a7-44fa-95dc-34c042275068)